### PR TITLE
add 'list_ingressroute_for_all_namespaces' method

### PR DESCRIPTION
**Pull request by Priyankasaggu11929**
#### What type of PR is this?
/kind cleanup
/kind documentation

#### What this PR does / why we need it:

As mentioned in the [linked issue](https://github.com/kubernetes-client/python/issues/1388), there is a suggestion to implement a  `list_ingressroute_for_all_namespaces ` method, which is meant for listing all the `ingressroutes` resources across all the namespaces. This PR adds changes to implement the same.

This new method, `list_ingressroute_for_all_namespaces ` is analogous to executing `kubectl  get ingressroutes --all-namespaces` command.

Besides, implementing the above method, the PR also include changes for demonstrating creation & deletion of k8s namespace resources using `dynamic client`. The methods implemented are `create_namespace` & `deleted_namespace`.

*Output of the script:*

Screenshot from 2021-05-06 01-08-55

#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes-client/python/issues/1388

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: examples/dynamic-client/namespaced_custom_resource.py [hunk 1]
```diff
@@ -21,10 +21,44 @@
...
+            )
+        )
+
+def create_namespace(namespace_api, name):
+    namespace_manifest = {
+        "apiVersion": "v1",
+        "kind": "Namespace",
+        "metadata": {"name": name, "resourceversion": "v1"},
+    }
+    namespace = namespace_api.create(body=namespace_manifest)
```
Comment (yliaog) [inline examples/dynamic-client/namespaced_custom_resource.py:57]: namespace is not  used, do you need to assign to it?

File: examples/dynamic-client/namespaced_custom_resource.py [hunk 2]
```diff
@@ -21,10 +21,44 @@
...
+
+
+def delete_namespace(namespace_api, name):
+    namespace_deleted = namespace_api.delete(name=name)
```
Comment (yliaog) [inline examples/dynamic-client/namespaced_custom_resource.py:61]: namespace_deleted is not used, do you need to assign to it?

Comment (Priyankasaggu11929) [reply examples/dynamic-client/namespaced_custom_resource.py]: Thank you for pointing. No, the variable `namespace_deleted` is not required.

I've removed it.

Comment (Priyankasaggu11929) [reply examples/dynamic-client/namespaced_custom_resource.py]: Thank you for pointing. No, the variable `namespace` is not required. I've removed it.

**Comment by yliaog** [comment]
please squash the commits, otherwise looks good.

**Comment by Priyankasaggu11929** [comment]
Thank you. I squash the commits.

**Pull request merged**